sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/UnsafeWriter.java

```diff
@@ -132,16 +132,17 @@ private void writeUnalignedBytes(
 
   public final void write(int ordinal, CalendarInterval input) {
     // grow the global buffer before writing data.
-    grow(16);
+    grow(24);
 
-    // Write the months and microseconds fields of Interval to the variable length portion.
+    // Write the months, days and microseconds fields of Interval to the variable length portion.
     Platform.putLong(getBuffer(), cursor(), input.months);
```

Comment:
we write `months` as long to align with word(8 bytes) boundary. Now we have an extra int in the interval, so padding is not needed now.